sel/src/Sel/SecretKey/AuthenticatedEncryption.hs

```diff
@@ -135,15 +134,38 @@ secretKeyFromByteString bytestring =
   if BS.length bytestring < fromIntegral cryptoSecretboxKeyBytes
     then Nothing
     else unsafeDupablePerformIO $
-      BS.unsafeUseAsCStringLen bytestring $ \(outsideSecretKeyPtr, _) -> do
-        secretKeyForeignPtr <-
-          BS.mallocByteString @CChar (fromIntegral cryptoSecretboxKeyBytes)
-        Foreign.withForeignPtr secretKeyForeignPtr $ \secretKeyPtr ->
-          Foreign.copyArray
+      BS.unsafeUseAsCStringLen bytestring $ \(outsideSecretKeyPtr, _) ->
+        fmap Just $
+          newSecretKeyWith $ \secretKeyPtr ->
```

Comment:
This doesn't change any behavior, right? If it's just moving things around then it's fine.

Reply:
Unless I've missed something the old code was just unwrapping bytestring, does the same allocation routine, then copying bytes. Hence the common function.